src/shared/container/providers/CacheProvider/implementations/RedisCacheProvider.ts

```diff
@@ -0,0 +1,40 @@
+import Redis, { Redis as IRedis } from 'ioredis'
+import cacheConfig from '@config/cache'
+import ICacheProvider from '../models/ICacheProvider'
+
+export default class RedisCachePRovider implements ICacheProvider {
```

Comment:
Acho que tem um typo aqui:
```suggestion
export default class RedisCacheProvider implements ICacheProvider {
```